policy-controller/k8s/status/src/index.rs

```diff
@@ -0,0 +1,246 @@
+use crate::{
+    http_route::{ParentReference, RouteBinding},
+    resource_id::ResourceId,
+};
+use ahash::{AHashMap as HashMap, AHashSet as HashSet};
+use chrono::offset::Utc;
+use linkerd_policy_controller_k8s_api::{self as k8s, gateway, ResourceExt};
+use parking_lot::RwLock;
+use std::{collections::hash_map::Entry, sync::Arc};
+use tokio::sync::mpsc::{UnboundedReceiver, UnboundedSender};
+
+const POLICY_API_GROUP: &str = "policy.linkerd.io";
+const POLICY_API_VERSION: &str = "policy.linkerd.io/v1alpha1";
+const STATUS_CONTROLLER_NAME: &str = "policy.linkerd.io/status-controller";
+
+pub type SharedIndex = Arc<RwLock<Index>>;
+
+pub struct Controller {
+    client: k8s::Client,
+    index: SharedIndex,
+    updates: UnboundedReceiver<Update>,
+}
+
+pub struct Index {
+    updates: UnboundedSender<Update>,
+
```

Comment:
do we need this additional space?

Reply:
We don't, but we occasionally introduce newlines in other structs when there is a clear separation of fields in the struct. I did it here because `updates` serves a purpose very different from the `http_routes` and `servers` fields (the actually index part of `Index`).

It seems to be a style we use elsewhere in Rust code which is why I've done that here.

Reply:
yeah, i think it's fine to do this.